presto-cassandra/src/main/java/io/prestosql/plugin/cassandra/util/CassandraCqlUtils.java

```diff
@@ -121,4 +129,27 @@ public static Select selectDistinctFrom(CassandraTableHandle tableHandle, List<C
     {
         return from(select(columns).distinct(), tableHandle);
     }
+
+    public static String getWhereClause(List<CassandraPartition> partitions, String clusteringKeyPredicates)
+    {
+        List<String> conditions = new ArrayList<>();
+        for (CassandraPartition part : partitions) {
+            conditions.add(part.getPartitionId());
+        }
+        if (!clusteringKeyPredicates.isEmpty()) {
+            conditions.add(clusteringKeyPredicates);
+        }
+        return " WHERE " + String.join(" AND ", conditions);
+    }
+
+    public static String deleteFrom(CassandraTableHandle handle)
+    {
+        Delete delete = QueryBuilder.delete().from(handle.getSchemaName(), handle.getTableName());
+        Optional<List<CassandraPartition>> opt = handle.getPartitions();
+        List<CassandraPartition> partitions = opt.orElse(Collections.emptyList());
+        String clause = getWhereClause(partitions, handle.getClusteringKeyPredicates());
+        String cql = delete.getQueryString().replace(";", clause);
+        log.debug("Created CQL: %s", cql);
```

Comment:
Please move logging to caller side to focus the actual logic in util class.

Reply:
Got it.